programs/util.c

```diff
@@ -633,6 +638,247 @@ const char* UTIL_getFileExtension(const char* infilename)
    return extension;
 }
 
+static int pathnameHas2Dots(const char *pathname)
+{
+    return NULL != strstr(pathname, "..");
+}
+
+int UTIL_isFileNameValidForMirroredOutput(const char *filename)
+{
+    return !pathnameHas2Dots(filename);
+}
+
+
+#define DIR_DEFAULT_MODE 0755
+static mode_t getDirMode(const char *dirName)
+{
+    struct stat st;
+    int ret = UTIL_getDirectoryStat(dirName, &st);
+    if (!ret) {
+        UTIL_DISPLAY("zstd: failed to get DIR stats %s: %s\n", dirName, strerror(errno));
+        return DIR_DEFAULT_MODE;
+    }
+    return st.st_mode;
+}
+
+static int makeDir(const char *dir, mode_t mode)
+{
+    int ret = mkdir(dir, mode);
+    if (ret != 0) {
+        if (errno == EEXIST)
+            return 0;
+        UTIL_DISPLAY("zstd: failed to create DIR %s: %s\n", dir, strerror(errno));
+    }
+    return ret;
+}
+
+void UTIL_convertPathnameToDirName(char *pathname)
+{
+    int len = 0;
+    char* pos = NULL;
+    // get dir name from pathname similar to 'dirname'
```

Comment:
`C90` comment style is limited to `/*  (...)  */`

Reply:
done.